src/services/SessionService.js

```diff
@@ -12,23 +10,25 @@ export default {
   },
 
   getPartner() {
-    const user = UserService.getUser();
-    const session = this.currentSession.data || {};
+    return UserService.getUser().then(user => {
+      const session = this.currentSession.data || {};
 
-    if (user.isVolunteer) {
-      return session.student;
-    }
-    return session.volunteer;
+      if (user.isVolunteer) {
+        return session.student;
+      }
+      return session.volunteer;
+    });
   },
```

Comment:
Where in our code do we call this function? Is it really necessary?

Reply:
I believe we used to use this [in the chat](https://github.com/UPchieve/web/blob/master/src/views/SessionView/SessionChat.vue) in order to figure out things like the name of the other person in the session and what profile picture to show (student or volunteer profile) etc. doesn't seem like we are anymore though.